plugins/otter-pro/inc/plugins/class-block-conditions.php

```diff
@@ -149,6 +149,36 @@ public function evaluate_condition( $bool, $condition, $visibility ) {
 			}
 		}
 
+		if ( 'wooCategory' === $condition['type'] && class_exists( 'WooCommerce' ) ) {
+			if ( isset( $condition['categories'] ) ) {
+				if ( $visibility ) {
+					return $this->has_product_category( $condition['categories'] );
+				} else {
+					return ! $this->has_product_category( $condition['categories'] );
+				}
+			}
+		}
+
+		if ( 'wooTag' === $condition['type'] && class_exists( 'WooCommerce' ) ) {
+			if ( isset( $condition['tags'] ) ) {
+				if ( $visibility ) {
+					return $this->has_product_tag( $condition['tags'] );
+				} else {
+					return ! $this->has_product_tag( $condition['tags'] );
+				}
+			}
+		}
+
+		if ( 'wooAttribute' === $condition['type'] && class_exists( 'WooCommerce' ) ) {
+			if ( isset( $condition['attribute'] ) ) {
```

Comment:
I'd move this `if` condition to `has_product_attribute`. You're not using `$condition['attribute']` anywhere.

Reply:
We don't even want to run `has_product_attribute` method if `$condition['attribute']` isn't defined and move on the the next condition, that's why we check as that indicates condition wasn't setup completely by the user so we skip the check.

Reply:
I'm going to do a larger write up about this in the slack channel.